tfhe/src/integer/bigint/i256.rs

```diff
@@ -0,0 +1,77 @@
+use crate::core_crypto::prelude::CastFrom;
+
+pub type I256 = super::static_signed::StaticSignedBigInt<4>;
+
+impl CastFrom<crate::integer::U256> for I256 {
+    fn cast_from(input: crate::integer::U256) -> Self {
+        Self(input.0)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_const() {
+        assert_eq!(I256::BITS, 256);
+    }
+
+    #[test]
+    fn test_i128_convertion() {
+        let input = I256::from(-1i128);
+        assert_eq!(input.0, [u64::MAX, u64::MAX, u64::MAX, u64::MAX]);
+        assert_eq!(i128::cast_from(input), -1i128);
+
+        // 0b11111111...1000...00
+        // input is 64 0s in MSB, 64 1s in LSB
+        let input = i128::from(i64::MIN) << 1;
+        let big = I256::from(input);
+        assert_eq!(
+            big.0,
+            [(i64::MIN << 1) as u64, u64::MAX, u64::MAX, u64::MAX]
+        );
+        assert_eq!(i128::cast_from(big), input);
+    }
+
+    #[test]
+    fn test_add_wrap_around() {
+        assert_eq!(I256::MAX + I256::from(1i32), I256::MIN);
+        assert_eq!(I256::MIN + I256::from(-1i32), I256::MAX);
+    }
+
+    #[test]
+    fn test_sub_wrap_around() {
+        assert_eq!(I256::MIN - I256::from(1i32), I256::MAX);
+        assert_eq!(I256::MAX - I256::from(-1i32), I256::MIN);
+    }
+
+    #[test]
+    fn test_add() {
+        assert_eq!(I256::from(-1i32) + I256::ONE, I256::ZERO);
+    }
+
+    #[test]
+    fn test_sub() {
+        assert_eq!(I256::ONE - I256::ONE, I256::ZERO);
+        assert_eq!(I256::ONE - I256::from(-1i32), I256::TWO);
+    }
+
+    #[test]
+    fn test_shl_limits() {
+        assert_eq!(I256::ONE << 256u32, I256::ONE << (256 % I256::BITS));
+        assert_eq!(I256::ONE << 257u32, I256::ONE << (257 % I256::BITS));
+
+        // We aim to have same behaviour as rust native types
+        assert_eq!(1i128.wrapping_shl(128), 1i128 << (128 % i128::BITS));
+        assert_eq!(1i128.wrapping_shl(129), 1i128 << (129 % i128::BITS));
+    }
+
+    #[test]
+    fn test_shl() {
+        let input = i64::MIN as i128;
+        let a = I256::from(input);
+
+        assert_eq!(a << 1u32, I256::from(input << 1));
+    }
+}
```

Comment:
can we add tests for all ops with python examples given some of the potential issues referenced in other comments ?